gnd/src/main/java/com/google/android/gnd/persistence/local/room/entity/FeatureEntity.java

```diff
@@ -134,10 +168,33 @@ public static Feature toFeature(FeatureEntity featureEntity, Project project) {
       return builder.build();
     }
 
+    if (featureEntity.getPolygonVertices() != null) {
+      PolygonFeature.Builder builder =
+          PolygonFeature.builder().setVertices(stringToList(featureEntity.getPolygonVertices()));
+      fillFeature(builder, featureEntity, project);
+      return builder.build();
+    }
+
     throw new LocalDataConsistencyException(
         "No geometry data found in feature " + featureEntity.getId());
   }
 
+  public static ImmutableList<Point> stringToList(String vertices) {
+    if (vertices == null) {
+      Timber.d("vertices are null");
+      return null;
+    }
+    Gson gson = new Gson();
+    ArrayList<ArrayList<Double>> verticesArray =
+        gson.fromJson(vertices, new TypeToken<ArrayList<ArrayList<Double>>>(){}.getType());
```

Comment:
What happens if the vertices are empty or malformed? Is there an exception you can catch to log and return a default value?